docs/elementExtensions.md

```diff
@@ -0,0 +1,73 @@
+
+A host element can call the `load_element_extension()` function to load one specific extension, or `load_all_extensions_for_element()` to load everything that is available.  These are defined in the freeform `prairielearn` module.
+
+Loading extension Python scripts returns a named tuple of all globally defined functions and variables.  Loading all extensions will return a dictionary mapping the extension name to its named tuple.  For example, if an extension were to define the following in their controller:
+
+```python
+def my_cool_function():
+    return "hello, world!"
+```
+
+The host element could then call this by running the following:
+
+```python
+import prairielearn as pl
+
+def render(element_html, data):
+    extension = pl.load_element_extensions(data, 'extension_name')
+    contents = extension.my_cool_function()
+    return contents
+```
+
+This small example above will render `"hello world!"` to the question page.  Note that no order is guaranteed when loading all extensions with `load_all_extensions_for_element()`.
+
+### Extension Dependencies
+
+Like how questions and elements may require client-side assets, extensions may also require client-side JavaScript and CSS.  The different properties are summarized here:
+
+Property | Description
+--- | ---
+`coreStyles` | The styles required by this extension, relative to `[PrairieLearn directory]/public/stylesheets`.
+`coreScripts` | The scripts required by this extension, relative to `[PrairieLearn directory]/public/javascripts`.
+`nodeModulesStyles` | The styles required by this extension, relative to `[PrairieLearn directory]/node_modules`.
+`nodeModulesScripts` | The scripts required by this extension, relative to `[PrairieLearn directory]/node_modules`.
+`elementStyles` | The styles required by this element relative to the extension's directory,`[course directory]/elementExtensions/element-name/extension-name`.
+`elementScripts` | The scripts required by this element relative to the extension's directory, `[course directory]/elementExtensions/element-name/extension-name`.
+`clientFilesCourseStyles` | The styles required by this extension relative to `[course directory]/clientFilesCourse`.
+`clientFilesCourseScripts` | The scripts required by this extension relative to `[course directory]/clientFilesCourse`.
+
+Note that all client-side extension assets are always loaded, regardless of whether their Python controller was loaded or not.
+
+### Other Client Files
+
+Other files available to the client may also be loaded, such as images or any downloadable content.  These client files should be placed in `clientFilesExtension` in the extension directory, and the full URL to that folder is given to the host extension in `data['options']['client_files_extensions_url'][extension_name]`.  If this path is needed by the extension itself, it may be passed as an argument to a defined extension function.
```

Comment:
I think the documentation for developing a question element should include a reference back here or at the very least a note that extension data exists in `options`

https://github.com/PrairieLearn/PrairieLearn/blob/master/docs/question.md#question-serverpy